compiler/rustc_incremental/src/persist/load.rs

```diff
@@ -235,3 +195,35 @@ pub fn load_query_result_cache(sess: &Session) -> Option<OnDiskCache<'_>> {
         _ => Some(OnDiskCache::new_empty(sess.source_map())),
     }
 }
+
+/// Setups the dependency graph by loading an existing graph from disk and set up streaming of a
+/// new graph to an incremental session directory.
+pub fn setup_dep_graph(
+    sess: &Session,
+    crate_name: Symbol,
+    stable_crate_id: StableCrateId,
+) -> Result<DepGraph, ErrorGuaranteed> {
+    // `load_dep_graph` can only be called after `prepare_session_directory`.
+    prepare_session_directory(sess, crate_name, stable_crate_id)?;
+
+    let res = sess.opts.build_dep_graph().then(|| load_dep_graph(sess));
+
+    if sess.opts.incremental.is_some() {
+        sess.time("incr_comp_garbage_collect_session_directories", || {
```

Comment:
There does seem to be a small performance hit which is probably this no longer being done in parallel.

Reply:
looks like noise to me on the graph of the regressed perf tests

Reply:
It does look like the time `incr_comp_garbage_collect_session_directories` takes in practice is quite small, so it's probably just noise.

Reply:
Probably a relative/absolute graph thingy again. It looks small and maybe not worth looking into, but unlikely to be noise.

![image](https://github.com/rust-lang/rust/assets/247183/819d0a2a-d25c-46ba-b0e4-6ab9ebdfc079)

Reply:
I was referring to the wall time regression in the initial perf run. The instruction count isn't very useful as this PR involves parallelism.